f3read.c

```diff
@@ -379,6 +386,12 @@ static void iterate_files(const char *path, const long *files,
 		tot_overwritten += stats.secs_overwritten;
 		tot_size += stats.bytes_read;
 		and_read_all = and_read_all && stats.read_all;
+		if (stats.secs_corrupted && delete) {
```

Comment:
The full expression must be `delete && (stats.secs_corrupted || tot_changed || tot_overwritten)`